manifests/config/global/jenkins_config.pp

```diff
@@ -0,0 +1,24 @@
+# This private class is called from jenkins::config to manage Jenkins' config.xml
+#
+class jenkins::config::global::jenkins_config(
+  $localstatedir           = $::jenkins::localstatedir,
+  $user                    = $::jenkins::user,
+  $group                   = $::jenkins::group,
+  $jenkins_config_template = $::jenkins::jenkins_config_template,
+) {
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  $config_file = "${localstatedir}/config.xml"
```

Comment:
This could be rewritten as `$config_file = "${::jenkins::localstatedir}/config.xml"` and the class parameter removed.  This should be true for all of the parameters currently on this class.